pkg/graveler/settings/manager.go

```diff
@@ -0,0 +1,125 @@
+package settings
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/treeverse/lakefs/pkg/block"
+	"github.com/treeverse/lakefs/pkg/block/adapter"
+	"github.com/treeverse/lakefs/pkg/graveler"
+	"github.com/treeverse/lakefs/pkg/logging"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+)
+
+const (
+	settingsSuffixTemplate = "/%s/settings/%s.json"
+)
+
+// Manager is a key-value store for Graveler repository-level settings.
+// Each setting is stored under a key, and it can be any proto.Message.
+// A setting is saved as an object in the backing storage, so the performance of save/get is bound by the
+// performance of a storage put/get respectively.
+type Manager struct {
+	refManager                  graveler.RefManager
+	branchLock                  graveler.BranchLocker
+	blockAdapter                block.Adapter
+	committedBlockStoragePrefix string
+}
+
+// Save persists the given proto.Message under the given repository and key.
+func (m *Manager) Save(ctx context.Context, repositoryID graveler.RepositoryID, key string, message proto.Message) error {
+	repo, err := m.refManager.GetRepository(ctx, repositoryID)
+	if err != nil {
+		return err
+	}
+	messageBytes, err := proto.Marshal(message)
+	if err != nil {
+		return err
+	}
+	err = m.blockAdapter.Put(ctx, block.ObjectPointer{
+		StorageNamespace: string(repo.StorageNamespace),
+		Identifier:       fmt.Sprintf(settingsSuffixTemplate, m.committedBlockStoragePrefix, key),
```

Comment:
Identifier starts with a slash? Not again :scream:

Reply:
Fixed